amy/workshops/models.py

```diff
@@ -2876,6 +2876,32 @@ class CommonRequest(SecondaryEmailMixin, models.Model):
         verbose_name="Department/School/Library affiliation (if applicable)",
     )
 
+    MEMBER_AFFILIATION_CHOICES = (
+        ("yes", "Yes"),
+        ("no", "No"),
+        ("unsure", "I'm not sure"),
+    )
+    member_affiliation = models.CharField(
+        max_length=STR_MED,
+        null=False,
+        default="no",
+        blank=True,
+        choices=MEMBER_AFFILIATION_CHOICES,
+        verbose_name="Are you affiliated with a Carpentries member organization?",
+        help_text="This may be the same as your institution above, "
+        "or another institution.",
+    )
+    member_code = models.CharField(
```

Comment:
What are the values to be used here? Can you provide some samples?

What is the "secrecy level" of this code, can this leak to third parties and cause harm?

Reply:
These values are manually created by using a short name for the institution followed by two digits, e.g. `alpha55` for Alpha Organization, or `seu78` for Some Example University (SEU). The number of characters in the name part of the code can vary - the longest code I've seen is 9 characters total, but the corresponding DB field on Membership model permits up to 40.
For membership renewals, the trailing digits are changed, with no pattern. 

This code is only shared by us with designated member contacts, but as they will pass it on to those booking workshops/training there is a risk of it leaking that way. If a malicious actor got hold of a code they would theoretically be able to book workshops/training seats with it without the member contact knowing (unless the member contact was carefully validating the numbers in their quarterly emails).
But note that sometimes members will donate or share workshops/seats to/with other institutions, so just checking the code matches the institution in this form won't work